### ISSUE TITLE
Support for jest debug options (detectOpenHandles, logHeapUsage)

### ISSUE THREAD

**Issue by mchl18** [nrwl/nx#4621]
Hi, I am really wondering why there only is a small subset of jest parameters supported. Is there any good reason not to?

## Description
It would be very helpful if it was possible to add support for jests `detectOpenHandles` or `logHeapUsage`  options in order to debug memory leaks in tests.

## Motivation
We are working on a very large nx-workspace with ~20 modules and 7 apps. Recently our tests started failing with ` Allocation failed - JavaScript heap out of memory` and it would be great to be able to log these things without having to run jest directly, because in doing so our coverage reporting broke.

## Suggested Implementation
Basically like it is here: https://github.com/nrwl/nx/issues/1217#issuecomment-479856176

**jest.builder.js**

```
    var config = {
+         detectOpenHandles: options.detectOpenHandles, 
+         logHeapUsage: options.logHeapUsage,
          watch: options.watch,
          coverage: options.codeCoverage,
          bail: options.bail,
```
